Mage.Sets/src/mage/cards/g/GiantTrapDoorSpider.java

```diff
@@ -0,0 +1,116 @@
+ *  ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ *  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+ *  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *  The views and conclusions contained in the software and documentation are those of the
+ *  authors and should not be interpreted as representing official policies, either expressed
+ *  or implied, of BetaSteward_at_googlemail.com.
+ */
+package mage.cards.g;
+
+import java.util.UUID;
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.common.SimpleActivatedAbility;
+import mage.abilities.costs.common.TapSourceCost;
+import mage.abilities.costs.mana.ManaCostsImpl;
+import mage.abilities.effects.Effect;
+import mage.abilities.effects.common.ExileSourceEffect;
+import mage.abilities.effects.common.ExileTargetEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Zone;
+import mage.filter.common.FilterAttackingCreature;
+import mage.game.Game;
+import mage.game.combat.CombatGroup;
+import mage.game.permanent.Permanent;
+import mage.target.common.TargetCreaturePermanent;
+
+/**
+ *
+ * @author TheElk801
+ */
+public class GiantTrapDoorSpider extends CardImpl {
+
+    public GiantTrapDoorSpider(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{1}{R}{G}");
+
+        this.subtype.add("Spider");
+        this.power = new MageInt(2);
+        this.toughness = new MageInt(3);
+
+        // {1}{R}{G}, {tap}: Exile Giant Trap Door Spider and target creature without flying that's attacking you.
+        Ability ability = new SimpleActivatedAbility(Zone.BATTLEFIELD, new ExileSourceEffect(), new ManaCostsImpl("{1}{R}{G}"));
+        ability.addCost(new TapSourceCost());
+        Effect effect = new ExileTargetEffect();
+        effect.setText("Exile Giant Trap Door Spider and target creature that's attacking you.");
+        ability.addEffect(new ExileTargetEffect());
+        ability.addTarget(new TargetCreaturePermanent(new HuntingKavuFilter()));
+        this.addAbility(ability);
+    }
+
+    public GiantTrapDoorSpider(final GiantTrapDoorSpider card) {
+        super(card);
+    }
+
+    @Override
+    public GiantTrapDoorSpider copy() {
+        return new GiantTrapDoorSpider(this);
+    }
+}
+
+class HuntingKavuFilter extends FilterAttackingCreature {
+
+    public HuntingKavuFilter() {
+        super("creature that's attacking you");
+    }
+
+    public HuntingKavuFilter(final HuntingKavuFilter filter) {
+        super(filter);
+    }
+
+    @Override
+    public HuntingKavuFilter copy() {
+        return new HuntingKavuFilter(this);
+    }
+
+    @Override
+    public boolean match(Permanent permanent, UUID sourceId, UUID playerId, Game game) {
+        if (!super.match(permanent, sourceId, playerId, game)) {
```

Comment:
You can shorten this filter by using this statement:
return super.match(permanent, sourceId, playerId, game)
   && permanent.isAttacking()  // to prevent unneccessary combat checking if not attacking
   && playerId.equals(game.getCombat().getDefenderId(permanent.getId()));